coralogix/clientset/apikeys-client.go

```diff
@@ -0,0 +1,66 @@
+package clientset
+
+import (
+	"context"
+	apikeys "terraform-provider-coralogix/coralogix/clientset/grpc/apikeys"
+)
+
+type ApikeysClient struct {
+	callPropertiesCreator *CallPropertiesCreator
+}
+
+func (t ApikeysClient) CreateApiKey(ctx context.Context, req *apikeys.CreateApiKeyRequest) (*apikeys.CreateApiKeyResponse, error) {
+	callProperties, err := t.callPropertiesCreator.GetCallProperties(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	conn := callProperties.Connection
+	defer conn.Close()
+	client := apikeys.NewApiKeysServiceClient(conn)
+
+	return client.CreateApiKey(callProperties.Ctx, req, callProperties.CallOptions...)
```

Comment:
got the impression that you can wrap up those lines into a single method  instead of copying on each one

Reply:
I don't think we are going to safe any line, The defer needs to in each function, the rest is error handling

Reply:
ok.